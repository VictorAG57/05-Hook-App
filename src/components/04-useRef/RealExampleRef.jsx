import React, { Fragment, useState } from 'react';
import '../02-useEffect/effects.css'
import MultipleCustomHook from '../03-examples/MultipleCustomHook';

const RealExampleRef = () => {


    const [show, setShow] = useState(false)

    return (
        <Fragment>
            <h1>Real custom hook</h1>
            <hr />

            { show && <MultipleCustomHook></MultipleCustomHook> }

            <button 
                className="btn btn-primary"
                onClick={
                    ()=> setShow( !show )
                }
                >
                    Show
            </button>
            
        </Fragment>
    )
}

export default RealExampleRef
