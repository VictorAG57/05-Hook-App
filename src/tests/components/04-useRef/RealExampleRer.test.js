import { shallow } from "enzyme";
import RealExampleRef from "../../../components/04-useRef/RealExampleRef";

describe('Pruebas en <RealExampleRef />', () => {
    
    const wrapper = shallow(<RealExampleRef></RealExampleRef>);
    test('Debe de mostrar el componente', () => {

        expect( wrapper ).toMatchSnapshot();
        
    });

    test('Debe de mostrar el componente', () => {

        // Simulamos un click en el boton del componente <RealExampleRef></RealExampleRef>
        // ya que al dar click, mostrara la informaci√≥n que ay dentro de 
        // <MultipleCustomHook></MultipleCustomHook>, por ello la necesidad de 
        // simular el click.
        wrapper.find('button').simulate('click');

        // Verificamos que si existe el componene <MultipleCustomHook></MultipleCustomHook>
        // nos muestre un true, eso significa que se simulo correctamente el click.
        // Esto a su vez significa que useRef funciona correctamente.
        expect( wrapper.find('MultipleCustomHook').exists()).toBe( true );    
    });
});
