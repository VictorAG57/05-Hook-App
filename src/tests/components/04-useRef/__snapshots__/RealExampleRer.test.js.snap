// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <RealExampleRef /> Debe de mostrar el componente 1`] = `
<Fragment>
  <h1>
    Real custom hook
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Show
  </button>
</Fragment>
`;
